lib/jnpr/junos/utils/start_shell.py

```diff
@@ -112,10 +124,11 @@ def run(self, command, this=_SHELL_PROMPT):
         # run the command and capture the output
         self.send(command)
         got = ''.join(self.wait_for(this))
-
-        # use $? to get the exit code of the command
-        self.send('echo $?')
-        rc = ''.join(self.wait_for(this))
+        rc = ''
+        if re.search(r'(%|>|#)\s?$', got) is not None:
+            # use $? to get the exit code of the command
+            self.send('echo $?')
+            rc = ''.join(self.wait_for(this))
```

Comment:
if `this != _SHELL_PROMPT` then this line will end up waiting until the `timeout` value.
Instead, I believe this line should be `rc = ''.join(self.wait_for(_SHELL_PROMPT))`